corehq/apps/reports/models.py

```diff
@@ -1097,3 +1099,21 @@ def remove_custom_export(cls, domain, export_id):
     def clear_caches(self):
         super(HQGroupExportConfiguration, self).clear_caches()
         self.by_domain.clear(self.__class__, self.domain)
+
+
+class ReportsSidebarOrdering(models.Model):
+    domain = models.CharField(
+        max_length=256,
+        null=False,
+        blank=False,
+        unique=True
+    )
+    config = JSONField(
+        default=list,
+        help_text=(
+            "An array of arrays. Each array represents a heading in the sidebar navigation. "
+            "The first item in each array is a string, which will be the title of the heading. The second item in "
+            "the array is another array, each item of which is the name of a report class. Each of these reports "
+            "will be listed under the given heading in the sidebar nav."
+        )
+    )
```

Comment:
yeah was thinking something exactly like that. You could also add a property called `wrapped_config` which would return the jsonobject version of the config which might be nicer to work with in the calling code as well.

unfortunately, we don't have the equivalent of `JsonField(schema=MyDocument)` though that would be pretty nice for a future PID...